### PR TITLE
switch to jbuilder

### PR DESCRIPTION
**Pull request by vbmithr**

### CODE REVIEW

File: ipaddr.install [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  "_build/install/default/lib/ipaddr/macaddr/macaddr.cmi" {"macaddr/macaddr.cmi"}
+  "_build/install/default/lib/ipaddr/unix/ipaddr_unix.cmxs" {"unix/ipaddr_unix.cmxs"}
+  "_build/install/default/lib/ipaddr/unix/ipaddr_unix.a" {"unix/ipaddr_unix.a"}
+  "_build/install/default/lib/ipaddr/unix/ipaddr_unix.cmxa" {"unix/ipaddr_unix.cmxa"}
+  "_build/install/default/lib/ipaddr/unix/ipaddr_unix.cma" {"unix/ipaddr_unix.cma"}
+  "_build/install/default/lib/ipaddr/unix/ipaddr_unix.mli" {"unix/ipaddr_unix.mli"}
+  "_build/install/default/lib/ipaddr/unix/ipaddr_unix.cmti" {"unix/ipaddr_unix.cmti"}
+  "_build/install/default/lib/ipaddr/unix/ipaddr_unix.cmt" {"unix/ipaddr_unix.cmt"}
+  "_build/install/default/lib/ipaddr/unix/ipaddr_unix.cmx" {"unix/ipaddr_unix.cmx"}
+  "_build/install/default/lib/ipaddr/unix/ipaddr_unix.cmi" {"unix/ipaddr_unix.cmi"}
```
Comment (dsheets) [inline ipaddr.install]: The `ipaddr_unix` files are optional depending on presence of `base-unix`.

**Comment by vbmithr** [comment]
Should be good now.

**Comment by avsm** [comment]
CI failing due to missing opam depends:

```
# Error: I couldn't find 'ocaml-migrate-parsetree.driver-main'.
# I need this library in order to use ppx rewriters.
```

**Comment by avsm** [comment]
I've bumped the minimum ocaml version to 4.03.0, and pushed some jbuilder fixes to meet the standards in https://mirage.io/wiki/packaging

**Comment by dsheets** [comment]
Why is 4.03.0 required? Also, why move to 2.8.0? Was some feature added?

**Comment by rgrinberg** [comment]
@dsheets  recent ppx_sexp_conv (the only ones that work using omp) require 4.03.0 at a minimum.

**Comment by avsm** [comment]
> Also, why move to 2.8.0? Was some feature added?

Nothing in the library interfaces has changed, but the entire build system has, so it felt appropriate to bump the middle version number... I can go back to 2.7.x if you prefer.  

I'll push a note to the changelog explaining the 4.03/sexp_conv dependency.

**Comment by dsheets** [comment]
Sure. Thanks!

**Pull request merged**